packages/mui-utils/src/mergeProps.ts

```diff
@@ -0,0 +1,14 @@
+export default function mergeProps<T extends { className?: string } & Record<string, unknown>>(
```

Comment:
When I saw this method as `mergeProps` I was expecting that it would take care of merging `className` correctly, merging `sx` etc. Should we make sure it does this? Otherwise the name is misleading

Reply:
Good point. My purpose is not deep merging but to resolve props if a default exists.

These are other options I can think of:
- `resolveProps`
- `prepareProps`
- `shallowMergeProps`

> Note, we should not merge the className in this PR since it changes the behavior and might break some app.

Reply:
Let's call it `resolveProps` and add  JSDocs of what is expected that it would do.

Reply:
Done!